### PR TITLE
A few reMarkable QoL fixes

### PR DESCRIPTION
**Pull request by NiLuJe**
* Enable AutoSuspend (#6769)
* Better power events handling (#6676)
* Enable HW inversion, because I don't see why it wouldn't work ;p.

Many thanks to the actual testers and their feedback in the respective issues :).

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/6772)

### CODE REVIEW

**Comment by ezdiy** [comment]
hwinvert: `Hardware        : Freescale i.MX6 SoloLite (Device Tree)` should be fine

**Comment by NiLuJe** [comment]
Especially since rM are probably the only ones actually using the latest NXP+EPDC kernel tree ;p.

**Review by Frenzie — APPROVED** [review]
lgtm

File: frontend/device/generic/device.lua [hunk 1]
```diff
@@ -289,20 +291,30 @@ function Device:onPowerEvent(ev)
...
-          --       This is critical on Kobos (c.f., #3936), where it might still be on from KSM or Nickel,
-          --       without us being aware of it (i.e., wifi_was_on still unset or false),
-          --       because suspend will at best fail, and at worst deadlock the system if Wi-Fi is on,
-          --       regardless of who enabled it!
-          if network_manager:isWifiOn() then
-              network_manager:releaseIP()
-              network_manager:turnOffWifi()
+          -- NOTE: This side of the check needs to be laxer, some platforms can handle Wi-Fi without WifiManager ;).
+          if self:hasWifiToggle() then
+            local network_manager = require("ui/network/manager")
```
Comment (poire-z) [inline frontend/device/generic/device.lua:308]: Strange indentation around there (already there before your changes - but fixing it won't increase the diff :)

**Comment by gbyl** [comment]
@NiLuJe Just letting you know that this bfe6820b58c2e8456db073f87f2c64b2002b8a24 actually broke Night Mode. It appears that whatever mechanism was triggering HW inversion before is running at the same time. So with that new commit, once you tap on Night Mode, it inverts twice – effectively undoing the inversion once you tap away to hide the menu.

**Comment by NiLuJe** [comment]
@gbyl: I assume you're confusing a black flash along the way with a double-inversion, and this just means that HW inversion is borked for some strange reason.

Oh, well.

**Comment by NiLuJe** [comment]
@gbyl: Just to double check, if you run

`/mnt/root/koreader/fbink -MmH "Hello World"`, is it displayed white on black?

(Feel free to fix KOReader's actual install path on the rM ^^).

**Comment by gbyl** [comment]
@NiLuJe That's correct! "Hello World' is displayed white on black. rM's path is `home/root/koreader` I always wondered why it was never changed here https://github.com/koreader/koreader/blob/39faa8b860325c3c18372a4b24e50e1a1b407a8b/frontend/device/remarkable/device.lua#L20

**Comment by NiLuJe** [comment]
Huh, then HW inversion *does* work. Grmbl.

**Comment by NiLuJe** [comment]
Or it's buggy in *some* weird circumstances, which, err, we've unfortunately seen before ;p.

**Comment by NiLuJe** [comment]
Next question: Since I've kind of lost track, is this with KOReader running @ 8bpp or at xochitl's stock 16bpp?

**Comment by gbyl** [comment]
hmm I don't know how to check that

**Comment by NiLuJe** [comment]
That would have been mentioned in the FBInk console output ;).

**Comment by gbyl** [comment]
It says `8bpp`

**Comment by NiLuJe** [comment]
Then, assuming the KOReader install is sane, that's... strange ;).

----

Re-reading your message, though, it *vaguely* looks like the PxP ignoring the flags passed to it after a (short) while in your case, which is similar to some of the weird issues I've mentioned before.

Just to confirm:

* Start with nightmode disabled.
* Open menu
* Tick Night mode
* Screen refreshes (the flash may be wonky or stroby, that's normal, nightmode is terrible).
* Screen is properly completely inverted
* Close menu (again, possibly stuttery/strobey flash, nightmode sucks)
* Screen is no longer completely inverted ?!

----

If that's what happens, can you try doing the same thing after unticking "Flash buttons and menu items", and ticking "Avoid mandatory black flashes in UI" in Gear > Screen >E-ink settings

Are the results similar? Or do you perhaps end up with only bits of screen inverted?

**Comment by gbyl** [comment]
Actually… I forgot that I already changed `canHWInvert = yes,` to `canHWInvert = no,` before I ran the Hello World message.

On an unmodified v2020.09-57-g39faa8b, `/mnt/root/koreader/fbink -MmH "Hello World"` prints black on white.

**Comment by gbyl** [comment]
> Just to confirm:
> 
> * Start with nightmode disabled.
> * Open menu
> * Tick Night mode
> * Screen refreshes (the flash may be wonky or stroby, that's normal, nightmode is terrible).
> * Screen is properly completely inverted
> * Close menu (again, possibly stuttery/strobey flash, nightmode sucks)
> * Screen is now  completely NOT inverted ?!

This is a perfect step-by-step description of what happens.

> If that's what happens, can you try doing the same thing after unticking "Flash buttons and menu items", and ticking "Avoid mandatory black flashes in UI" in Gear > Screen >E-ink settings
> Are the results similar? Or do you perhaps end up with only bits of screen inverted?

The results are the same.

**Comment by NiLuJe** [comment]
`canHWInvert` should have zero incidence on FBInk (the two are completely orthogonal), so, I'm going to go ahead and assume, that, yes, HWInvert is wonky as hell :D.

**Comment by gbyl** [comment]
Yeah, no big deal. Our time is better spent implementing more essential features like Wi-Fi management.

**Pull request merged**